services/ui-src/src/utils/reports/entities.ts

```diff
@@ -141,6 +141,195 @@ export const getFormattedEntityData = (
           (method: EntityShape) => method.value
         );
 
+        const notAnsweredOptional = "Not answered; optional";
+
+        const findNestedData = (keys: any[], keyMatchText: string) => {
+          // compliance frequency and type
+          const dataKey: string = keys.find((key: string) =>
+            key.endsWith(keyMatchText)
+          );
+          if (dataKey) {
+            if (Array.isArray(plan[dataKey])) {
+              const value = plan[dataKey]?.[0]?.value;
+              return value ?? notAnsweredOptional;
+            }
+            return `${plan[dataKey]}` ?? notAnsweredOptional;
+          }
+          return;
+        };
+
+        const indexOfGeomapping = analysisMethodsUsed.indexOf("Geomapping");
+        if (indexOfGeomapping >= 0) {
+          const geomappingInfo: any[] = [];
+          const geomappingKeys: any = planKeys.filter((key: string) =>
+            key.includes("_geomappingComplianceFrequency")
+          );
```

Comment:
only look at geomapping related keys (since some of the other nested fields will have the same suffixes)